lib/agenda/define.ts

```diff
@@ -12,18 +27,18 @@ const debug = createDebugger('agenda:define');
  * @param options options for job to run
  * @param processor function to be called to run actual job
  */
-export const define = function(this: Agenda, name: string, options: any, processor: () => void) {
-  if (!processor) {
-    processor = options;
+export const define = function(this: Agenda, name: string, options: DefineOptions | (() => void), processor?: () => void) {
+  if (processor === undefined) {
+    processor = options as (() => void);
```

Comment:
`"as Processor"`

Reply:
you could ppbl also improve the check.. e.g. by checking `typeof options === 'function'` ..